cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -74,28 +75,40 @@ func (pc *PoolConfig) expandPool() error {
 }
 
 // addGroupToPool adds a raid group to the csp
-func addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
+func (pc *PoolConfig) addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
 	for _, cspcRaidGroup := range cspcPoolSpec.RaidGroups {
+		validGroup := true
 		cspcRaidGroup := cspcRaidGroup
 		if !isRaidGroupPresentOnCSP(&cspcRaidGroup, csp) {
 			if cspcRaidGroup.Type == "" {
 				cspcRaidGroup.Type = cspcPoolSpec.PoolConfig.DefaultRaidGroupType
 			}
-			csp.Spec.RaidGroups = append(csp.Spec.RaidGroups, cspcRaidGroup)
+			for _, bd := range cspcRaidGroup.BlockDevices {
+				err := pc.isBDUsable(bd.BlockDeviceName)
+				if err != nil {
+					glog.Errorf("could not use bd %s for expanding pool "+
+						"%s:%s", bd.BlockDeviceName, csp.Name, err.Error())
+					validGroup = false
+					break
+				}
+			}
+			if validGroup {
+				csp.Spec.RaidGroups = append(csp.Spec.RaidGroups, cspcRaidGroup)
+			}
 		}
 	}
 	return csp
 }
 
 // expandStripedGroup adds newly added block devices to the striped
 // groups present on CSP
-func expandStripedGroup(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) {
+func (pc *PoolConfig) expandStripedGroup(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) {
 	for _, cspcGroup := range cspcPoolSpec.RaidGroups {
 		cspcGroup := cspcGroup
 		if getRaidGroupType(cspcGroup, cspcPoolSpec) != "stripe" || !isRaidGroupPresentOnCSP(&cspcGroup, csp) {
```

Comment:
lets bring out discussion on the type 'stripe'.. it would be great if you can make it as a single global constant without any worry to change later